plugin/hover.py

```diff
@@ -112,15 +112,12 @@ def diagnostics_content(self, diagnostics):
         by_severity = {}  # type: Dict[int, List[str]]
         for diagnostic in diagnostics:
             by_severity.setdefault(diagnostic.severity, []).append(self.format_diagnostic(diagnostic))
-        formatted = []
+        formatted = ""
```

Comment:
joining a list of strings is "significantly faster": https://stackoverflow.com/questions/3055477/how-slow-is-pythons-string-concatenation-vs-str-join